sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Result.scala

```diff
@@ -123,7 +144,7 @@ object ResultError {
   */
 final case class ResultNeedContract[A](
     acoid: ContractId,
-    resume: Option[FatContractInstance] => Result[A],
+    resume: Option[ContractAndIdValidator] => Result[A],
```

Comment:
Need to change this to a tri-state structure:
- Contract found
- Contract not found
- Contract found but invalid (cannot establish hash type)